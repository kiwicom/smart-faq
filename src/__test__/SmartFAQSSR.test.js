// @flow

import * as React from 'react';
import ReactDOMServer from 'react-dom/server';

import SmartFAQ from '../App';
import Translations from '../translations/locales-fallback/en-GB.json';

const user = {
  id: '1',
  email: 'joe.doe@example.com',
  firstname: 'Joe',
  lastname: 'Doe',
};

const chatConfig = {
  ORG_ID: '1234',
};

describe('SmartFAQ', () => {
  it('should work on the server', () => {
    const app = ReactDOMServer.renderToStaticMarkup(
      <SmartFAQ
        onClose={jest.fn()}
        onLogin={jest.fn()}
        onSocialLogin={jest.fn()}
        onLogout={jest.fn()}
        language="en"
        translations={Translations}
        emergencies={[]}
        user={user}
        route="/"
        loginToken="AAABBBCCC"
        simpleToken={null}
        enableChat={false}
        chatConfig={chatConfig}
        onAppWithOpenChatClose={jest.fn()}
      />,
    );
    expect(app.includes('Need help?')).toBe(true);
    expect(app).toMatchSnapshot();
  });

  it('should render faqs', () => {
    const app = ReactDOMServer.renderToStaticMarkup(
      <SmartFAQ
        onClose={jest.fn()}
        onLogin={jest.fn()}
        onSocialLogin={jest.fn()}
        onLogout={jest.fn()}
        language="en"
        translations={Translations}
        emergencies={[]}
        user={user}
        route="/faq/"
        loginToken="AAABBBCCC"
        simpleToken={null}
        enableChat={false}
        chatConfig={chatConfig}
        onAppWithOpenChatClose={jest.fn()}
      />,
    );
    expect(app.includes('Help')).toBe(true);
    expect(app).toMatchSnapshot();
  });
});
