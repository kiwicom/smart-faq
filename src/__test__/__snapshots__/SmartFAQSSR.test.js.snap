// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartFAQ should render faqs 1`] = `"<div class=\\"focus-trap-wrapper\\"><div aria-hidden=\\"true\\" class=\\"focus-trap-backdrop\\"></div><div tabindex=\\"0\\" class=\\"focus-trap\\"><div data-test=\\"SmartFAQHelp\\" class=\\"jsx-68938604 smartFAQ open\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1 shrink-to-fit=no\\" class=\\"jsx-68938604\\"/><meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" class=\\"jsx-68938604\\"/><link href=\\"https://fonts.googleapis.com/css?family=Roboto\\" rel=\\"stylesheet\\" class=\\"jsx-68938604\\"/><div class=\\"jsx-780438590 ContentPage\\"><div id=\\"SmartFAQ_header\\" class=\\"jsx-4290801200 header\\"><div class=\\"jsx-4290801200 HeaderFAQ\\"><div class=\\"jsx-2880528156 jsx-2367066345 loggedIn\\"><div class=\\"jsx-2880528156 jsx-2367066345 helpHeader\\">Help</div></div></div></div><div class=\\"jsx-780438590 Body\\"><div class=\\"jsx-780438590 FAQWrapper\\"><div class=\\"jsx-3319669865 scrollable-content\\"><div id=\\"SmartFAQ_Body\\" class=\\"jsx-780438590 FAQ\\"><div class=\\"jsx-1306298768 static-faq\\"><div class=\\"jsx-1306298768 static-faq-body\\"><div class=\\"jsx-1306298768 static-faq-search\\"></div><div class=\\"jsx-4051752212 wrapper\\"><img src=\\"https://images.kiwi.com/smart-faq/spinner@2x.png\\" alt=\\"Loading\\" class=\\"jsx-4051752212 spinner\\"/></div></div></div></div></div></div></div></div></div></div></div>"`;

exports[`SmartFAQ should work on the server 1`] = `"<div class=\\"focus-trap-wrapper\\"><div aria-hidden=\\"true\\" class=\\"focus-trap-backdrop\\"></div><div tabindex=\\"0\\" class=\\"focus-trap\\"><div data-test=\\"SmartFAQHelp\\" class=\\"jsx-68938604 smartFAQ open\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1 shrink-to-fit=no\\" class=\\"jsx-68938604\\"/><meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" class=\\"jsx-68938604\\"/><link href=\\"https://fonts.googleapis.com/css?family=Roboto\\" rel=\\"stylesheet\\" class=\\"jsx-68938604\\"/><div class=\\"jsx-3382029431 Intro\\"><div role=\\"button\\" tabindex=\\"-1\\" class=\\"jsx-477917638 close-icon\\"><svg fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\" height=\\"24px\\" width=\\"24px\\" viewBox=\\"0 0 24 24\\" style=\\"vertical-align:middle;color:#46515e\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></div><div class=\\"jsx-3382029431 picture\\"><img class=\\"Illustration__StyledImage-bPSPtN hiagLf\\" src=\\"//images.kiwi.com/illustrations/0x200/Help.png\\" srcSet=\\"//images.kiwi.com/illustrations/0x400/Help.png 2x\\" alt=\\"Help\\"/></div><div class=\\"jsx-3382029431 text\\"><p class=\\"jsx-3382029431 title\\">Need help?</p><p class=\\"Text__StyledText-inVtPV fMhFpH\\">We&#x27;re here for you. First, let&#x27;s narrow down your request.</p></div><div class=\\"jsx-3382029431 buttons\\"><div class=\\"jsx-3382029431 primary\\"><button data-cy=\\"btn-existent-booking\\" class=\\"jsx-3382029431\\">I have an existing booking</button></div><div class=\\"jsx-3382029431 secondary\\"><button data-cy=\\"btn-nonexistent-booking\\" class=\\"jsx-3382029431\\">I don&#x27;t have a booking</button></div></div><hr class=\\"jsx-3382029431 hr-line\\"/><div class=\\"jsx-3382029431 faqLink\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"/helpcenter\\" class=\\"jsx-3798246845 \\">Full FAQ site<span class=\\"jsx-3798246845 inline-icon\\"><div class=\\"jsx-3798246845 open-icon\\"><svg fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\" height=\\"16px\\" width=\\"16px\\" viewBox=\\"0 0 24 24\\" style=\\"vertical-align:middle;color:#171b1e\\"><path d=\\"M17.444 17.444H6.556V6.556H12V5H6.556C5.692 5 5 5.7 5 6.556v10.888C5 18.3 5.692 19 6.556 19h10.888C18.3 19 19 18.3 19 17.444V12h-1.556v5.444zM13.556 5v1.556h2.792L8.702 14.2 9.8 15.298l7.645-7.646v2.792H19V5h-5.444z\\"></path></svg></div></span></a></div></div></div></div></div>"`;
